### PR TITLE
Refactor `SelectionManager` to replace `Hashtable` with `Dictionary`

### PR DESCRIPTION
**Pull request by elachlan**
Related: #8143, #8357

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8664)

### CODE REVIEW

**Review by dmitrii-drobotov — APPROVED** [review]

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/SelectionManager.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ internal sealed class SelectionManager : IDisposable
         private Adorner _bodyAdorner;                       //used to track all body glyphs for each control
         private BehaviorService _behaviorService;           //ptr back to our BehaviorService
         private IServiceProvider _serviceProvider;          //standard service provider
-        private readonly Hashtable _componentToDesigner;    //used for quick look up of designers related to comps
+        private readonly Dictionary<IComponent, ControlDesigner> _componentToDesigner;    //used for quick look up of designers related to comps
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/SelectionManager.cs:24]:
```suggestion
        private readonly Dictionary<IComponent, ControlDesigner> _componentToDesigner;    // used for quick look up of designers related to components
```

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/SelectionManager.cs [hunk 2]
```diff
@@ -66,9 +66,9 @@ public SelectionManager(IServiceProvider serviceProvider, BehaviorService behavi
             _bodyAdorner = new Adorner();
             behaviorService.Adorners.Add(_bodyAdorner);
             behaviorService.Adorners.Add(_selectionAdorner); //adding this will cause the adorner to get setup with a ptr
-                                                            //to the beh.svc.
+                                                             //to the beh.svc.
```
Comment (lonitra) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/SelectionManager.cs:69]:
```suggestion
                                                             // to the behavior service.
```

**Review by lonitra — COMMENTED** [review]
Just a nit on the comments that were already there, otherwise LGTM

File: src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/SelectionManager.cs [hunk 3]
```diff
@@ -152,12 +152,11 @@ private void AddAllControlGlyphs(Control parent, ArrayList selComps, object prim
         /// <summary>
         ///  Recursive method that goes through and adds all the glyphs of every child to our global Adorner.
         /// </summary>
-        private void AddControlGlyphs(Control c, GlyphSelectionType selType)
+        private void AddControlGlyphs(Control control, GlyphSelectionType selType)
         {
-            ControlDesigner cd = (ControlDesigner)_componentToDesigner[c];
-            if (cd is not null)
+            if (_componentToDesigner.TryGetValue(control, out ControlDesigner controlDesigner))
```
Comment (dreddy-work) [inline src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/SelectionManager.cs:157]: Q: are we certain that null values were not inserted? Earlier we were checking for null value explicitly.

Comment (elachlan) [reply src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/Behavior/SelectionManager.cs]: Good idea to check for null anyway. We will go over this a bit more later when nullability annotations are added.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**